invokeai/app/services/object_serializer/object_serializer_forward_cache.py

```diff
@@ -0,0 +1,62 @@
+from queue import Queue
+from typing import TYPE_CHECKING, Optional, TypeVar
+
+from invokeai.app.services.object_serializer.object_serializer_base import ObjectSerializerBase
+
+T = TypeVar("T")
+
+if TYPE_CHECKING:
+    from invokeai.app.services.invoker import Invoker
+
+
+class ObjectSerializerForwardCache(ObjectSerializerBase[T]):
+    """Provides a simple forward cache for an underlying storage. The cache is LRU and has a maximum size."""
+
+    def __init__(self, underlying_storage: ObjectSerializerBase[T], max_cache_size: int = 20):
+        super().__init__()
+        self._underlying_storage = underlying_storage
+        self._cache: dict[str, T] = {}
+        self._cache_ids = Queue[str]()
+        self._max_cache_size = max_cache_size
+
+    def start(self, invoker: "Invoker") -> None:
+        self._invoker = invoker
+        start_op = getattr(self._underlying_storage, "start", None)
+        if callable(start_op):
+            start_op(invoker)
+
+    def stop(self, invoker: "Invoker") -> None:
+        self._invoker = invoker
+        stop_op = getattr(self._underlying_storage, "stop", None)
+        if callable(stop_op):
+            stop_op(invoker)
```

Comment:
My understanding - the original intent is twofold:
1. Make things services that can have startup and shutdown logic.
2. Provide access to other services via the start method, instead of needing to provide all deps explicitly in the `__init__` method.

I'm not familiar enough with architectural design to comment on whether this pattern is good or bad